client/landing/stepper/declarative-flow/internals/steps-repository/ecommerce-segmentation-survey/queries/use-survey-answers-query.ts

```diff
@@ -0,0 +1,35 @@
+import { useQuery } from '@tanstack/react-query';
+import { Answers } from 'calypso/components/survey-container/types';
+import wpcom from 'calypso/lib/wp';
+
+type SurveyAnswersQueryParams = {
+	surveyKey: string;
+};
+
+type SurveyAnswersResponse = {
+	question_key: string;
+	answer_key: string;
+}[];
+
+const mapSurveyAnswers = ( response: SurveyAnswersResponse ): Answers => {
+	return response.reduce( ( acc: Record< string, string[] >, { question_key, answer_key } ) => {
+		acc[ question_key ] = [ ...( acc[ question_key ] || [] ), answer_key ];
```

Comment:
Also, I'm open to the idea of converting underscore_cased properties to camelCase, if it makes typing easier in the long run.

Reply:
Agree. Grouping on the backend side would be better. Let's align with @JuanLucha before moving to that approach.

Reply:
> Also, I'm open to the idea of converting underscore_cased properties to camelCase, if it makes typing easier in the long run.

I don't think it would be necessary, we can easily map the response to match with the typescript types in camelCase, no need for changing it on the backend.

Reply:
Yeah, I just defined the response in snake_case to match the API response, but the actual `Answers` type is basically a Record<string, string[]>, so there is no snake_case property, just a map of question-to-answers keys.